docs/source/config_syntax.md

```diff
@@ -0,0 +1,187 @@
+
+The main benefits are threefold:
+
+- it provides good readability and usability by separating system parameter settings from the Python code.
+- it describes workflow at a relatively high level and allows for different low-level implementations.
+- learning paradigms at a higher level such as federated learning and AutoML can be decoupled from the component details.
+
+Content:
+
+- [A basic example](#a-basic-example)
+- [Syntax examples explained](#syntax-examples-explained)
+  - [`@` to interpolate with Python objects](#1--to-interpolate-with-python-objects)
+  - [`$` to evaluate as Python expressions](#2--to-evaluate-as-python-expressions)
+  - [`%` to interpolate with plain texts](#3--to-interpolate-with-plain-texts)
+  - [`_target_` (`_disabled_` and `_requires_`) to instantiate a Python object](#4-instantiate-a-python-object)
+- [The command line interface](#the-command-line-interface)
+- [Recommendations](#recommendations)
+
+## A basic example
+
+Components as part of a workflow can be specified using `JSON` or `YAML` syntax, for example, a network architecture
+definition could be stored in a `demo_config.json` file with the following content:
+
+```json
+{
+  "demo_net": {
+    "_target_": "monai.networks.nets.BasicUNet",
+    "spatial_dims": 3,
+    "in_channels": 1,
+    "out_channels": 2,
+    "features": [16, 16, 32, 32, 64, 64]
+  }
+}
+```
+
+or alternatively, in `YAML` format (`demo_config.yaml`):
+
+```yaml
+demo_net:
+  _target_: monai.networks.nets.BasicUNet
+  spatial_dims: 3
+  in_channels: 1
+  out_channels: 2
+  features:
+  - 16
+  - 16
+  - 32
+  - 32
+  - 64
+  - 64
+```
+
+The configuration parser can instantiate the component as a Python object:
+
+```py
+>>> from monai.bundle import ConfigParser
+>>> config = ConfigParser()
+>>> config.read_config("demo_config.json")
+>>> net = config.get_parsed_content("demo_net", instantiate=True)
+BasicUNet features: (16, 16, 32, 32, 64, 64).
+>>> print(type(net))
+<class 'monai.networks.nets.basic_unet.BasicUNet'>
+```
+
+or additionally, tune the input parameters then instantiate the component:
+
+```py
+>>> config["demo_net"]["features"] = [32, 32, 32, 64, 64, 64]
+>>> net = config.get_parsed_content("demo_net", instantiate=True)
+BasicUNet features: (32, 32, 32, 64, 64, 64).
+```
+
+For more details on the `ConfigParser` API, please see https://docs.monai.io/en/latest/bundle.html#config-parser.
+
+## Syntax examples explained
+
+A few characters and keywords are interpreted beyond the plain texts, here are examples of the syntax:
+
+### 1. `@` to interpolate with Python objects
+
+```json
+"@preprocessing#transforms#keys"
+```
+
+_Description:_ A reference to another configuration value defined at `preprocessing#transforms#keys`.
+where `#` indicates a sub-structure of this configuration file.
+
+```json
+"@preprocessing#1"
+```
+
+_Description:_ `1` is interpreted as an integer, which is used to index the `preprocessing` sub-structure.
```

Comment:
Suggest to describe 2 points: "list" and "index from 0", maybe:
which is used to index (count from `0`) the `preprocessing` sub-structure list.

Thanks.